src/interpreter.rs

```diff
@@ -212,6 +261,19 @@ where
         }
         Statement::Goto(line_number) => Action::Goto(*line_number),
         Statement::Gosub(line_number) => Action::Gosub(*line_number),
+        Statement::If(left_expression, relationship, right_expression, line_number) => {
+            if evaluate_if(
+                left_expression,
+                relationship,
+                right_expression,
+                line_number,
+                state,
+            )? {
+                Action::Goto(*line_number)
```

Comment:
What do you think about return the action from `evaluate_if`?

Reply:
then it's not exactly consistent with the other evaluate_* calls is it? I like seeing what `Action` are taken in the core loop of the state machine.

Reply:
Ah. True. I thought we always returned the `Action`.